### PR TITLE
Updating DNS powershell to new SDK

### PR DESCRIPTION
**Pull request by vladrai**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vladrai__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by hovsepm** [comment]
@azuresdkci retest this please

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -5873,7 +5873,6 @@
             <ComponentRef Id="cmp5B16E8141BC633DB9DEB2A9361AF4C71" />
             <ComponentRef Id="cmp24741C738CAC6E0595C4035CF8FE190A" />
             <ComponentRef Id="cmpB21325D18CF34CB53A2C33585CE3C81B" />
-            <ComponentRef Id="cmpF693C79DAAA8541CFC3919A5E8AFE1DC" />
             <ComponentRef Id="cmpBA345C2F015DD9066F6ACA7FAC093AC9" />
```
Comment (hovsepm) [inline setup/azurecmdfiles.wxi]: please revert this change.

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj [hunk 2]
```diff
@@ -72,11 +72,14 @@
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.27.306291202\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.1.0\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.1\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj]: Please revert this versions back to 2.1.0

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj [hunk 3]
```diff
@@ -72,11 +72,14 @@
...
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.1.0\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.3.1\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/Commands.ScenarioTests.ResourceManager.Common.csproj]: Please revert this back to 3.1.0

File: src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/packages.config [hunk 4]
```diff
@@ -3,10 +3,10 @@
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Common.Dependencies" version="1.0.0" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Resources" version="2.18.14-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.ResourceManager" version="1.1.1-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Resources" version="2.18.14-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.Framework" version="1.0.5945.28173-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.0.5945.28173-prerelease" targetFramework="net45" />
+  <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.3-preview" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Common/Commands.ScenarioTests.ResourceManager.Common/packages.config]: Please revert this back

File: src/ResourceManager/Dns/Commands.Dns/packages.config [hunk 5]
```diff
@@ -14,7 +14,7 @@
   <package id="Microsoft.Data.Services.Client" version="5.6.4" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.27.306291202" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.1" targetFramework="net45" />
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/packages.config]: back to 2.1.0 please

File: src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj [hunk 6]
```diff
@@ -109,6 +112,10 @@
     <Reference Include="Microsoft.Azure.Common.NetFramework">
       <HintPath>..\..\..\packages\Microsoft.Azure.Common.2.1.0\lib\net45\Microsoft.Azure.Common.NetFramework.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Azure.Test.HttpRecorder, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.6.3-preview\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj]: remove this completely. Product code should not depend on Test.HttpRecorder

File: src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj [hunk 7]
```diff
@@ -120,18 +127,26 @@
...
     </Reference>
     <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms, Version=2.27.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.27.306291202\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.1.0\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.1\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj]: Back to 2.1.0 please

File: src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj [hunk 8]
```diff
@@ -120,18 +127,26 @@
...
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.1.0\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Commands.Dns.csproj]: you need to sync csproj and packages.config. There is no Microsoft.Rest.ClientRuntime.Azure package dependency in packages.config. Do you really need it?

File: src/ResourceManager/Dns/Commands.Dns/Models/DnsClient.cs [hunk 9]
```diff
@@ -321,5 +449,51 @@ private static List<DnsRecordBase> GetPowerShellRecords(Management.Dns.Models.Re
...
+    //    private const string fileName = @"d:\scratch\pslog.txt";
+
+    //    public static void LogMesg(string s)
+    //    {
+    //        using (var file = File.AppendText(fileName))
+    //        {
+    //            file.WriteLine(s);
+    //        }
+    //    }
+    //}
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Models/DnsClient.cs]: remove commented code.

File: src/ResourceManager/Dns/Commands.Dns/Records/NewAzureDnsRescordSet.cs [hunk 10]
```diff
@@ -51,9 +51,12 @@ public class NewAzureDnsRecordSet : DnsBaseCmdlet
         [ValidateNotNullOrEmpty]
         public RecordType RecordType { get; set; }
 
-        [Alias("Tags")]
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Records/NewAzureDnsRescordSet.cs]: this would be a breaking change for customers. I think you should keep Tags Alias here for backward compatibility.

Comment (jtuliani) [inline src/ResourceManager/Dns/Commands.Dns/Records/NewAzureDnsRescordSet.cs]: The service is in Preview.  A breaking change like this is OK.

File: src/ResourceManager/Dns/Commands.Dns/Zones/GetAzureDnsZone.cs [hunk 11]
```diff
@@ -21,28 +21,21 @@ namespace Microsoft.Azure.Commands.Dns
...
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The full name of the zone (without a terminating dot).")]
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
-        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource group in which the zone exists.")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = ParameterSetResourceGroup, HelpMessage = "The resource group in which the zone exists.")]
         [ValidateNotNullOrEmpty]
         public string ResourceGroupName { get; set; }
 
-        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The single or multiple label suffix to search in the zone name.")]
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Zones/GetAzureDnsZone.cs]: Instead of removing you may give customers some time (say one release) to change their existing scripts. Just mark this parameter with following attribute [Obsolete("EndsWith parameter will be removed in an upcoming release", false)]

File: src/ResourceManager/Dns/Commands.Dns/Zones/RemoveAzureDnsZone.cs [hunk 12]
```diff
@@ -92,12 +86,42 @@ public override void ExecuteCmdlet()
...
+
+        public DnsZone GetDnsZoneHandleNonExistentZone(string zoneName, string resourceGroupName)
+        {
+            DnsZone retrievedZone = null;
+            try
+            {
+                if (zoneName.EndsWith("."))
+                {
+                    zoneName = zoneName.TrimEnd('.');
+                    this.WriteWarning(string.Format("Modifying zone name to remove terminating '.'.  Zone name used is \"{0}\".", zoneName));
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns/Zones/RemoveAzureDnsZone.cs]: should be moved to Resources

File: src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 13]
```diff
@@ -56,36 +56,62 @@
...
     </Reference>
-    <Reference Include="Microsoft.Azure.ResourceManager, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+    <Reference Include="Microsoft.Azure.Management.ResourceManager, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Resources.2.20.0-preview\lib\net40\Microsoft.Azure.ResourceManager.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.ResourceManager.1.1.1-preview\lib\net45\Microsoft.Azure.Management.ResourceManager.dll</HintPath>
+      <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Test.Framework">
+    <Reference Include="Microsoft.Azure.Test.Framework, Version=1.0.5486.28526, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: revert this line.

File: src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 14]
```diff
@@ -56,36 +56,62 @@
...
-      <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.0.5945.28173-prerelease\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Test.HttpRecorder.1.6.3-preview\lib\net45\Microsoft.Azure.Test.HttpRecorder.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory, Version=2.27.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.27.306291202\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.dll</HintPath>
+      <Private>True</Private>
+    </Reference>
+    <Reference Include="Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms">
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: please add Full assembly name

File: src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 15]
```diff
@@ -56,36 +56,62 @@
...
+      <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.27.306291202\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
+      <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.1.0\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.2.3.1\lib\net45\Microsoft.Rest.ClientRuntime.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: back to 2.1.0 please

File: src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj [hunk 16]
```diff
@@ -56,36 +56,62 @@
...
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.3.1.0\lib\net45\Microsoft.Rest.ClientRuntime.Azure.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Rest.ClientRuntime.Azure.Authentication, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.Authentication.2.2.4-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.Authentication.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.2.0-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
```
Comment (hovsepm) [inline src/ResourceManager/Dns/Commands.Dns.Test/Commands.Dns.Test.csproj]: please sync with your packages.config. There is mismatch in the library references.

File: src/ResourceManager/Dns/Commands.Dns/Zones/GetAzureDnsZone.cs (hunk 11 above)
Comment (vladrai) [inline src/ResourceManager/Dns/Commands.Dns/Zones/GetAzureDnsZone.cs]: We added resourceGroup name recently and then figured out that it's not needed so we want to remove it sooner.
